File: cpp/13.Roman_to_Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map <char,int> M{
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}, 
        };
        int sum = M[s.back()];
        for(int i =0 ;i<s.size()-1;i++){
            if(M[s[i]]<M[s[i+1]])
                sum -= M[s[i]];
            else
                sum += M[s[i]];
        }
        
        return sum;
    }
};

File: py/13. Roman to Integer.py
class Solution(object):
    def romanToInt(self, s):
        
        d={'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        l=list(str(s))
        k=len(l)
        i=0
        
        s=0
        if k==1:
            return d[l[0]]
        while i<k-1:
            j=i+1
          
            if d[l[i]]>d[l[j]]:
                s=s+d[l[i]]
                if i+2==k:
                    s=s+d[l[j]]
                i=i+1
            elif d[l[i]]<d[l[j]]:
                s=s+d[l[j]]-d[l[i]]
                i=i+2
                if i+1==k:
                    s=s+d[l[i]]
          
            else:
              
                if i+2==k:
                    s=s+2*(d[l[i]])
                    i=i+1
                else:
                    s=s+d[l[i]]
                    i=i+1
          
            
            
        return s
            
            
                
            
       
        
